subgraphs/convex/src/modules/Withdraw.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  Token,
+  Vault as VaultStore,
+  Withdraw as WithdrawTransaction,
+} from "../../generated/schema";
+import {
+  log,
+  BigInt,
+  Address,
+  ethereum,
+  BigDecimal,
+} from "@graphprotocol/graph-ts";
+import {
+  getOrCreateYieldAggregator,
+  getOrCreateUsageMetricsDailySnapshot,
+  getOrCreateUsageMetricsHourlySnapshot,
+} from "../common/initializer";
+import * as utils from "../common/utils";
+import { getUsdPricePerToken } from "../Prices";
+import * as constants from "../common/constants";
+import { ERC20 } from "../../generated/Booster/ERC20";
+import { Pool as PoolContract } from "../../generated/Booster/Pool";
+
+export function createWithdrawTransaction(
+  to: Address,
+  vaultId: string,
+  transaction: ethereum.Transaction,
+  block: ethereum.Block,
+  assetId: string,
+  amount: BigInt,
+  amountUSD: BigDecimal
+): WithdrawTransaction {
+  let withdrawTransactionId = "withdraw-" + transaction.hash.toHexString();
+
+  let withdrawTransaction = WithdrawTransaction.load(withdrawTransactionId);
+
+  if (!withdrawTransaction) {
+    withdrawTransaction = new WithdrawTransaction(withdrawTransactionId);
+
+    withdrawTransaction.vault = vaultId;
+    withdrawTransaction.protocol = constants.CONVEX_BOOSTER_ADDRESS.toHexString();
+
+    withdrawTransaction.to = to.toHexString();
+    withdrawTransaction.from = transaction.from.toHexString();
+
+    withdrawTransaction.hash = transaction.hash.toHexString();
+    withdrawTransaction.logIndex = transaction.index.toI32();
+
+    withdrawTransaction.asset = assetId;
+    withdrawTransaction.amount = amount;
+    withdrawTransaction.amountUSD = amountUSD;
+
+    withdrawTransaction.timestamp = block.timestamp;
+    withdrawTransaction.blockNumber = block.number;
+
+    withdrawTransaction.save();
+  }
+
+  return withdrawTransaction;
+}
+
+export function _Withdraw(

```

Comment:
I'd recommend naming these lowercase to differentiate from generated code. Also why underscore in the front?